### PR TITLE
Artifactory Upgrade to v7 and Update to install on OCP4

### PR DESCRIPTION
**Pull request by caggles**

### CODE REVIEW

**Comment by caggles** [comment]
for anyone looking this over (cough cough Jeff) - the new/working version of artifactory v7 is built from either everything.yaml or everything-editted.yaml (I believe both should work at the moment). It works as a fresh installation (not as an update yet) and does not have an HA version of postgres running under it. The other stuff (in artifactory-ha) is the updated set of scripts for automatically installing/upgrading the artifactory v6 - i'm keeping them because, if i can't get the helm chart to work to generate perfect installation scripts every time, i'll take the resulting scripts, edit them manually, and put them into the same format that we've used previously for v6.

**Comment by caggles** [comment]
hey @jefkel and @ShellyXueHan - after a billion years of waiting, this is finally ready for actual review :D :D :D

File: apps/artifactory/artifactory-ha/README.md [hunk 1]
```diff
@@ -1,124 +1,105 @@
...
-      * [License](#License)
-      * [Parameters](#Parameters)
-   * [Access Artifactory](#Access-Artifactory)
-<!--te-->
-
-## Architecture
-
-### Artifactory HA Architecture
-
-![image alt text](../images/artifactory-diagram.png)
```
Comment (ShellyXueHan) [inline apps/artifactory/artifactory-ha/README.md:21]: image not available in this branch

File: apps/artifactory/artifactory-ha/templates/patroni-deploy.yaml [hunk 2]
```diff
@@ -11,29 +11,46 @@ metadata:
...
-  app: ${DATABASE_NAME}${SUFFIX}
+  app: ${NAME}${SUFFIX}
   phase: deploy
-  app.kubernetes.io/instance: ${DATABASE_NAME}${SUFFIX}
+  app.kubernetes.io/instance: ${NAME}${SUFFIX}
   app.kubernetes.io/component: database
-  app.kubernetes.io/name: patroni  
+  app.kubernetes.io/name: patroni
   app.kubernetes.io/managed-by: template
-  app.kubernetes.io/version: "10"
```
Comment (ShellyXueHan) [inline apps/artifactory/artifactory-ha/templates/patroni-deploy.yaml:20]: not version 11?

File: apps/artifactory/artifactory-ha/templates/artifactory-config.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  hideUnauthorizedResources: true
+  oauthSettings:
+    persistUsers: true
+    allowUserToAccessProfile: true
+    enableIntegration: true
+    oauthProvidersSettings:
+      devhub-sso:
+        id: {{ sso_id }}
+        enabled: true
+        apiUrl: "https://sso.pathfinder.gov.bc.ca/auth/realms/devhub/protocol/openid-connect/userinfo"
```
Comment (ShellyXueHan) [inline apps/artifactory/artifactory-ha/templates/artifactory-config.yaml.j2:11]: better parameterize URL in vars.yaml/secrets/configmap for easier change

**Review by ShellyXueHan — APPROVED** [review]

**Pull request merged**